lib/codecs/src/encoding/mod.rs

```diff
@@ -50,23 +50,29 @@ impl From<std::io::Error> for Error {
     }
 }
 
-/// Configuration for building a `Framer`.
+/// Framing configuration.
 // Unfortunately, copying options of the nested enum variants is necessary
 // since `serde` doesn't allow `flatten`ing these:
 // https://github.com/serde-rs/serde/issues/1402.
-#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+#[configurable_component]
+#[derive(Clone, Debug, Eq, PartialEq)]
 #[serde(tag = "method", rename_all = "snake_case")]
 pub enum FramingConfig {
-    /// Configures the `BytesEncoder`.
+    /// Event data is not delimited at all.
+    ///
+    /// This implies that all data received in a single I/O operation will be treated as a single event.
```

Comment:
Hm, why does the framer on the _encoding_ side imply something about the data _received_? Did you mean to add this comment on the framer on the decoding side? If so, this used to be the behavior at the beginning. But it turned out unfeasible since the I/O boundaries can be pretty arbitrary for e.g. TCP and carry little intrinsic value. So we changed it to collect until the entire stream closes in that case

Reply:
Yeah, was just thinking about it backwards when I wrote the comment. I'll fix it.